### PR TITLE
Added documentation and submodule repo

### PR DESCRIPTION
**Pull request by alitariq4589**
- I have changed the documentation of github_webhook_integration to github_push_webhook_integration considering the fact that jenkins' built-in plugin can only be used to trigger job for push in the specified branch and not when pull request is created.
- I have added the documentation about using `Github Pull Request Builder` plugin and how it can be used to trigger the pipeline build when a pull request for merge is generated.
- Adding path of a package in .bashrc does not work in jenkins, so I have added documentation to add the package's path to $PATH variable in jenkins node configuration.
- Also I have added new submodule repository.

### CODE REVIEW

**Review by mmoizhussain — CHANGES_REQUESTED** [review]

File: jenkins_docs/Github_push_webhook.md [hunk 1]
```diff
@@ -92,4 +96,8 @@ For the sake of this documentation, I have created a simple repository called `j
       
 After this point each time the github repository is commited with a change, jenkins job will start the build and will also denote on the repository if the build has passed or failed (as can be seen in the below screenshot).
   
-![Screenshot from 2022-09-26 16-39-29](https://user-images.githubusercontent.com/99069972/192314484-a761bb98-5fbe-4303-9627-d42ff1f096e1.png)
+![Screenshot from 2022-09-26 16-39-29](https://user-images.githubusercontent.com/99069972/192314484-a761bb98-5fbe-4303-9627-d42ff1f096e1.png)
```
[the remark this answers is not in the document]
Comment (alitariq4589) [reply jenkins_docs/Github_push_webhook.md:99]: Okay

File: jenkins_docs/Building_Jenkins_github_repo.md [hunk 2]
```diff
@@ -21,4 +21,8 @@ After the above commands successfully completes execution, `jenkins.war` should
 `java -jar ~/jenkins/war/target/jenkins.war --httpPort=8080 #Considering jenkins repo is cloned at ~/`
 
 **After this process jenkins UI can be accessed using http://localhost:8080 in a browser and a password will be shown on terminal to login to jenkins first time**  
-After this, Jenkins UI will go through a very simple post installation process which one can configure according to his needs.
+After this, Jenkins UI will go through a very simple post installation process which one can configure according to his needs.  
+  
+<p align="center">
```
[the remark this answers is not in the document]
Comment (alitariq4589) [reply jenkins_docs/Building_Jenkins_github_repo.md]: I have reverted them back. I did not add a brief message with them when making changes so it was pretty hectic changing them again.

File: jenkins_docs/Creating_Jenkins_Node_on_LXC.md [hunk 3]
```diff
@@ -164,4 +164,8 @@ If no issue is encountered during this whole setup, jenkins will take us to the
   
 ![unnamed (14)](https://user-images.githubusercontent.com/99069972/190384770-a8f2fcff-081a-470a-913a-2982f943f16d.png)
```
[the remark this answers is not in the document]
Comment (alitariq4589) [reply jenkins_docs/Creating_Jenkins_Node_on_LXC.md:165]: Those are added from local directory. Btw I have tried to give local path to images on github markdown but the path does not resolve.

[the remark this answers is not in the document]
Comment (alitariq4589) [reply jenkins_docs/Creating_Jenkins_Node_on_LXC.md]: I have added relative paths of all the images in all documentations and the images are present in `doc_images`.

**Review by mmoizhussain — APPROVED** [review]

**Pull request merged**